java-checks-test-sources/spring-3.2/src/main/java/checks/WeakSSLContextCheckSample.java

```diff
@@ -0,0 +1,38 @@
+package checks;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.ssl.PropertiesSslBundle;
+import org.springframework.boot.autoconfigure.ssl.SslProperties;
+import org.springframework.boot.ssl.DefaultSslBundleRegistry;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class WeakSSLContextCheckSample {
+
+  public static final String TLSV_1_0 = "TLSv1.0";
+
+  @Autowired
+  public WeakSSLContextCheckSample(SslProperties props, DefaultSslBundleRegistry registry, Set<String> propsSet) {
+    props.getBundle().getJks().get("").getOptions().setEnabledProtocols(Set.of("TLSv1.1", "test", "TLSv1.0")); // Noncompliant [[sc=53;ec=72;secondary=18,18]] {{Change this code to use a stronger protocol.}}
```

Comment:
A workaround for avoiding the problem of not really being able to test secondaries very well is to split the line into multiple lines, one per secondary. This way you can be relatively precise, even though the way we test for secondaries in sonar-java isn't great.